weblate/static/editor/full.js

```diff
@@ -36,6 +36,34 @@
       submitForm({ target: this.$translationArea });
     });
 
+    /* Delete machinery results */
+    this.$editor.on("click", ".js-delete-machinery", (e) => {
+      var $el = $(e.target);
+
+      /* Delete Url dialog */
+      var $deleteEntriesDialog = null;
+      this.$editor.on("show.bs.modal", "#delete-url-modal", (e) => {
+        $deleteEntriesDialog = $(e.currentTarget);
+        $deleteEntriesDialog.find(".modal-body").html("");
+        var text = $el.parent().parent().data("raw").text;
+        var modalBody = this.machinery.renderDeleteUrls(text);
+        $deleteEntriesDialog.find(".modal-body").append(modalBody);
+      });
+
+      this.$editor.on("submit", ".delete-url-form", (e) => {
+        var $form = $(e.currentTarget);
+        var $deleteEntries = $form.find("input.form-check-input:checked");
+        $deleteEntriesDialog.modal("hide");
+
+        Object.entries($deleteEntries).forEach(([_, entry]) => {
+          if (typeof entry.id !== "undefined") {
+            this.removeTranslationEntry(entry.id);
+          }
+        });
+        return false;
+      });
```

Comment:
I believe here is the issue - the events are bind when the dialog is shown, so in case it is shown multiple times (dismissing it between), the `submit` handler is also triggered multiple times causing duplicate `DELETE` requests.

Reply:
after investigating a bit, I came to this solution.

When we dismiss the modal. the `.modal('hide')` returns to the method before the modal has actually been hidden. Here I assign the dialog as null so that when it reaches the submit handler, It doesn't do the `DELETE` request if the dialog is `null`.

It works fine now after multiple dismissals, only one `DELETE` request is sent. Please let me know wdyt?